### PR TITLE
Have default values in add database dialog input boxes

### LINKED ISSUE #12041: Add validation for SQLCMD variables in database reference dialog
Need to check for invalid characters and add $() around the variable if necessary. Show error message in Example usage section if name is invalid.

### PR DESCRIPTION
**Pull request by kisantia**
This adds default values for the database and server name and variable input boxes:
defaultValues2

Also update the example usage to handle if the sqlcmd variables are in the $() or not and show the example usage accordingly. If the input box value doesn't have $() around the variable, the example usage shows it as $(variable), and if the variable is already in the format $(variable), it doesn't get added. Also added validation for invalid sqlcmd variable names(fixes #12041)
image
image

### CODE REVIEW

File: extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts
Comment (Benjin) [inline extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts:357]: Can you make this into a GH issue and add the ID here?

File: extensions/sql-database-projects/src/common/utils.ts
Comment (Benjin) [inline extensions/sql-database-projects/src/common/utils.ts:176]: Maybe throw instead of returning empty string if it's not?

File: extensions/sql-database-projects/src/common/utils.ts
Comment (Benjin) [inline extensions/sql-database-projects/src/common/utils.ts:199]: If this throws instead of returning a boolean, you can give an explanation to the user as to why their variable name isn't valid.

File: extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/dialogs/addDatabaseReferenceDialog.ts:357]: I'll add this in this PR after #12186 is merged

File: extensions/sql-database-projects/src/common/constants.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/common/constants.ts:119]: SSDT allows it, but then there's an error when publishing if you have a variable name like $(ðĶƢ). Should we check for this or just let publishing fail?

File: extensions/sql-database-projects/src/common/utils.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/common/utils.ts]: I removed this comment since I added it before I figured out how the validation stuff would work.

File: extensions/sql-database-projects/src/common/utils.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/common/utils.ts:198]: created https://github.com/microsoft/azuredatastudio/issues/12231. For now, this is just the same as what SSDT does for invalid variable names

**Review by udeeshagautam — APPROVED** [review]
(re signing off my connection got interrupted)

Looks good. I might be missing some edge cases, will test with your change tomorrow.. but don't be blocked on me..

File: extensions/sql-database-projects/src/common/constants.ts
Comment (Benjin) [reply extensions/sql-database-projects/src/common/constants.ts:123]: I think just let publishing fail; if someone hits it and cares, they can submit a dacfx bug.  This way, we don't have to come back and change logic in db projects if that happens and we fix it.

The main reason I asked was that maybe localizing "OtherServer" would signal to users that we encourage them to use localized variable names which, since they're unicode, would fail during publish.  I think maybe we shouldn't localize these two.

**Review by Benjin — APPROVED** [review]

File: extensions/sql-database-projects/src/common/constants.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/common/constants.ts:123]: makes sense. I'll change them to not be localized

**Review by SakshiS-harma — APPROVED** [review]

File: extensions/sql-database-projects/src/test/dialogs/addDatabaseReferenceDialog.test.ts
Comment (kisantia) [reply extensions/sql-database-projects/src/test/dialogs/addDatabaseReferenceDialog.test.ts:58]: oops that's from when I was trying to figure out why the test was failing

**Pull request merged**
